lib/index.js

```diff
@@ -442,10 +443,37 @@ function invert(mat) {
     return derationalize(result);
 }
 
+/**
+ * Applies a given function over the matrix, elementwise. Similar to Array.map()
+ * The supplied function is provided 4 arguments:
+ * the current element,
+ * the row index,
+ * the col index,
+ * the matrix.
+ *
+ * @param mat input matrix
+ * @returns matrix of same dimensions with values altered by the function.
+ */
+function map(mat, func) {
+    const s = size(mat);
+    const result = [];
+    for (let i = 0; i < s[0]; i++) {
+        if(s[1]) {
```

Comment:
I would do `Array.isArray(mat[i])` here instead. Again there is nothing like which one is the right way. Were you going for performance?

Reply:
Good idea! I'll commit the change.